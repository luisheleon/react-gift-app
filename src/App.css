* {
  font-family: Helvetica, sans-serif;
}

body {
  padding: 60px;
}

.newCategoryForm {
  margin-bottom: 1rem;
}

.labelError {
  color: red;
  font-size: 15px;
  margin: 1rem;
  text-align: center;
  padding-top: auto;
}
