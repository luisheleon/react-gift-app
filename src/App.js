import React, { useState } from "react";
import AddCategory from "./components/AddCategory";
import CssBaseline from "@mui/material/CssBaseline";
import "./App.css";
import GifGrid from "./components/GifGrid";

function App({ defaultCategory = ["avatar", "goku"] }) {
  const [categories, setCategories] = useState(defaultCategory);

  return (
    <React.Fragment>
      <CssBaseline />
      <h1 id="title">GifExpertApp</h1>

      <hr />

      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </React.Fragment>
  );
}

export default App;
