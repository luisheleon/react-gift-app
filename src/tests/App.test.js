import App from "../App";
import { shallow } from "enzyme";

describe("Test <App />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  test("should create a snapshop", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should return title", () => {
    const title = "GifExpertApp";
    const htmlTitle = wrapper.find("#title").text().trim();
    const hr = wrapper.exists("hr");
    expect(htmlTitle).toBe(title);
    expect(hr).toBe(true);
  });
});
