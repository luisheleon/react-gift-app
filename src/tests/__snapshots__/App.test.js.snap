// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <App /> should create a snapshop 1`] = `
<Fragment>
  <CssBaseline />
  <h1
    id="title"
  >
    GifExpertApp
  </h1>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="avatar"
      key="avatar"
    />
    <GifGrid
      category="goku"
      key="goku"
    />
  </ol>
</Fragment>
`;
